### PR TITLE
INVGOV-13632 /notifications add more fields for client

### PR DESCRIPTION
**Pull request by klauselhai**
https://dev.invotra.com/?url=https://raw.githubusercontent.com/invotra/api/INVGOV-13633.SWAGGER-5.20/api.swagger.yaml

https://io1dev.atlassian.net/browse/INVGOV-13633

### CODE REVIEW

**Review by sam-ludlow — CHANGES_REQUESTED** [review]
Think we should use the existing term "target", all 3 types can be put together.

target: {
    target_type: 'node / comment / user' (enum)
    target_entity: {...} (node / comment / user' object)
}

Have a look in swagger at "discriminator". There needs to be a way for schema validation to determine what the type is.

I think we may have to create 3 schemas for each type that just have the type and ref to schema e.g:

{
    target_type: 'user' (use an enum with a single value)
    target_entity: $ref: '#/components/schemas/user-schema'
}

File: api.swagger.yaml [hunk 1]
```diff
@@ -12218,6 +12218,22 @@ components:
           description: The subject of the notification
           type: string
           example: mentioned you in a post
+        target:
```
Comment (sam-ludlow) [inline api.swagger.yaml:12221]: Have a look in swagger at "discriminator". There needs to be a way for schema validation to determine what the type is.

**Review by sam-ludlow — CHANGES_REQUESTED** [review]

**Review by sam-ludlow — CHANGES_REQUESTED** [review]
We need facets and filters on the notifications EP see https://io1dev.atlassian.net/browse/INVGOV-13632?focusedCommentId=302417

**Comment by MireilleHelle** [comment]
Closing task as we’re now focusing on building only API gaps for which Invotra Consulting have a direct need.

**Pull request closed without merging**